### DISCUSSION TITLE
wishlist item: mounted  /configs/.user_config

### DISCUSSION THREAD

**Discussion by virmaior** [gtxaspec/wz_mini_hacks#374] [Development]
Can we mount a camera configuration file `/configs/.user_config` before iCamera etc load?
if so where in the init.d sequence would it go?

This seems like a useful step both for wyze app free plug in card and go *and* for debugging changes to the configuration where rather than bricking the camera we just pull out the SD card to fix.

**Comment by kohrar** [comment]
This should be possible. From my understanding, `/configs` is mounted by the firmware-provided `/etc/init.d/rcS script` (which wz_mini_hacks copies to `SD_ROOT/wz_mini/tmp/.storage/rcS` and later edits as part of the startup sequence). You'll have to make a startup script that adds your `.user_config` bind mount after `/configs` get mounted.

One way you can do this is by modify [`07inject`](https://github.com/gtxaspec/wz_mini_hacks/blob/master/SD_ROOT/wz_mini/etc/init.d/S07inject) and adding something like the following, which appends the bind mount call at the end of the rcS script.
```
echo "mount --bind /opt/wz_mini/<somepath>/user_config /configs/.user_config &" >> /opt/wz_mini/tmp/.storage/rcS
```

**Reply by virmaior** [reply]
working on it. putting the mount code into 07inject and adding it to the end of rcS  is too late (possibly because my cameras technically still have WyzeHacks). Going to have to use sed to make sure it goes in above `/opt/wz_mini/tmp/.storage/app_init.sh &
`

**Comment by virmaior** [comment]
adding:
```
slap_path="/opt/wz_mini/slapin"
if [ -f "$slap_path/user_config" ]
then
        echo "found slap-in config"
        if [ ! -f "$slap_path/slapped_config" ]
        then
                sed -i -e '/mount -t jffs2 \/dev\/mtdblock6 \/configs/a\' -e 'mount --bind '$slap_path'/user_config /configs/.user_config' /opt/wz_mini/tmp/.storage/rcS
                
                echo "mount --bind $slap_path/user_config /configs/.user_config &" >> /opt/wz_mini/tmp/.storage/rcS
                touch "$slap_path/slapped_config"
                echo "loaded slap-in config"
        else
                echo "skipped due to indicator"
        fi
fi

```
creates a mount that enables us to load a configuration file from the SD card.

The `        if [ ! -f "$slap_path/slapped_config" ]` conditions makes it so that it will use the one on the camera itself upon reboot (I'm doing this for testing purposes).
We can do the same to override the ssid / ssidpassword if we want.

**Reply by mainmachine** [reply]
Have you started a branch or fork to track this? I'd like to help contribute and it looks like you've put a bit of work into this already so I'd like to align our efforts.

**Reply by virmaior** [reply]
I haven't made it into a fork at this point.
My proof of concept is just the above code put into `07inject`

the next steps I was going to work on (though completely open to other solutions):
1. add code at the end of boot chain to delete `"$slap_path/slapped_config"` to indicate a successful boot (and thus keep using the mounted version)
2. apply the same sort of mount to either the entire directory or to the ssid items
3. If to the entire directory have it setup to check if there are any files in the default mount not present in slapin and if so to copy them to slapin so it's a complete mount

It takes a while to work on this because I usually work while my wife sleeps, but we use the camera to track a flying squirrel ... who is usually awake when my wife sleeps ... so it's hard to find a good time to work on it.

**Reply by virmaior** [reply]
@mainmachine  whatever I had setup may have led to the complete clobber of the entire /configs directory. that camera is now permanently not linked to Wyze (lost *all* configuration info) but it now works on wz_mini (after quite some effort to recover it).

**Reply by endertable** [reply]
> @mainmachine whatever I had setup may have led to the complete clobber of the entire /configs directory. that camera is now permanently not linked to Wyze (lost _all_ configuration info) but it now works on wz_mini (after quite some effort to recover it).

wz_mini makes backups of all config files, no?  Or were they overwritten?

**Reply by virmaior** [reply]
@endertable the directory where wyze backs them up was empty (by the time I figured out it does). I'm not sure why. Since then I've made manual copies for .user_config and .product_config for all the other cameras. ...  was able to get it to record, do wz_mini, rtsp, etc. but it ain't connecting to the wyze app anymore.

Something strange was happening where the /configs directory wasn't mounting and neither was /media/mmc. the camera was also off by the difference between my time zone and gmt (probably from the loss of the config file). When I did get mount `/configs` it was empty so I had to copy over info from another camera (same settings so the only problem is the keys).

**Reply by endertable** [reply]
That bites.  One last ditch effort:  If it is a V3 cam, you can try to mount /dev/mtdblock4 and it will have a copy of .product_config in there named something like:  MD5.a2a8f3ba160c1be3ba2c0a3b66a41f99.config

**Reply by virmaior** [reply]
@endertable  thanks for the tip. I can't seem to mount /dev/mtdblock4 using squashfs or not specifying the type. 
Is it only mountable up to a certain spot in the boot process?

**Reply by endertable** [reply]
On all my cams, when I mount it, post boot, it mounts as a vfat?  But it just mounts without typing a fs type, e.g.:

mount /dev/mtdblock4 /mnt

Wyze started doing this "backup" process in more recent FW (circa early 2022) so my guess is you have an older version.  I reckon the best way to check is if you have /system/init/config_repair.sh on this cam?  If not, then you may have an older FW and therefore not have the backup on that partition. :(

**Reply by virmaior** [reply]
mine are still on 4.36.2.5 so nothing there... 
well, it's another motive to never use the app again (actually we now prefer the html page I made for multi-camera view -- even though it's only 2fps -- it's a lot better being able to see multiple angles at once).
